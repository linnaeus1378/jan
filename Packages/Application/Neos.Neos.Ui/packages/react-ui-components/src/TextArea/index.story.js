import React from 'react';
import {storiesOf, action} from '@storybook/react';
import {withKnobs, boolean} from '@storybook/addon-knobs';
import {StoryWrapper} from './../_lib/storyUtils';
import TextArea from '.';

storiesOf('TextArea', module)
    .addDecorator(withKnobs)
    .addWithInfo(
        'default',
        () => {
            const isValid = boolean('Is valid?', true);
            return (
                <StoryWrapper>
                    <TextArea
                        validationErrors={isValid ? null : ['This input is invalid']}
                        placeholder={isValid ? 'Valid input' : 'Invalid input'}
                        onChange={action('onChange')}
                        onFocus={action('onFocus')}
                        onBlur={action('onBlur')}
                        highlight={boolean('Highlight', false)}
                        />
                </StoryWrapper>
            );
        },
        {inline: true}
    );
