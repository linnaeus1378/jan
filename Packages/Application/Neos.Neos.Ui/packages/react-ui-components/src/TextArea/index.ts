import {themr} from '@friendsofreactjs/react-css-themr';
import identifiers from '../identifiers';
import style from './style.css';
import TextArea from './textArea';

const ThemedTextArea = themr(identifiers.textArea, style)(TextArea);

export default ThemedTextArea;
