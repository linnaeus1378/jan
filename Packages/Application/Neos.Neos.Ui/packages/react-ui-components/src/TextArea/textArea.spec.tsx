import React from 'react';
import {mount} from 'enzyme';
import toJson from 'enzyme-to-json';
import TextareaAutosize from 'react-textarea-autosize';

import TextArea, {TextAreaProps} from './textArea';

describe('<TextArea/>', () => {
    const props: TextAreaProps = {
        theme: {
            textArea: 'textAreaClassName',
            'textArea--disabled': 'disabledClassName',
        }
    };

    it('should render correctly.', () => {
        const wrapper = mount(<TextArea {...props} />);

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should allow the propagation of "className" with the "className" prop.', () => {
        const wrapper = mount(<TextArea {...props} className="fooClassName"/>);
        const input = wrapper.find(TextareaAutosize);
        expect(input.prop('className')).toContain('fooClassName');
    });

    it('should call the passed "onChange" prop with the value of the input when changing it.', () => {
        const onChange = jest.fn();
        const wrapper = mount(<TextArea {...props} onChange={onChange}/>);
        const input = wrapper.find(TextareaAutosize);

        input.simulate('change', {
            target: {
                value: 'my value'
            }
        });

        expect(onChange.mock.calls.length).toBe(1);
        expect(onChange.mock.calls[0][0]).toBe('my value');
    });
});
