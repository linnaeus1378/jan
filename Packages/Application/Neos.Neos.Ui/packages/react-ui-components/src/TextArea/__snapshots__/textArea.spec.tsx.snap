// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextArea/> should render correctly. 1`] = `
<clickOutside(TextArea)
  theme={
    Object {
      "textArea": "textAreaClassName",
      "textArea--disabled": "disabledClassName",
    }
  }
>
  <TextArea
    expandedRows={6}
    minRows={2}
    theme={
      Object {
        "textArea": "textAreaClassName",
        "textArea--disabled": "disabledClassName",
      }
    }
  >
    <ForwardRef(TextareaAutosize)
      aria-disabled="false"
      aria-multiline="true"
      className="textAreaClassName"
      minRows={2}
      onChange={[Function]}
      onClick={[Function]}
      role="textbox"
    >
      <textarea
        aria-disabled="false"
        aria-multiline="true"
        className="textAreaClassName"
        onChange={[Function]}
        onClick={[Function]}
        role="textbox"
      />
    </ForwardRef(TextareaAutosize)>
  </TextArea>
</clickOutside(TextArea)>
`;
